sentinel1_reader/sentinel1_reader.py

```diff
@@ -0,0 +1,331 @@
+    '''
+    Convert doppler poly1d to LUT2d.
+
+    Parameters
+    ----------
+    doppler_poly1d : poly1d
+        Poly1d object to be convereted.
+    starting_slant_range : float
+        Starting slant range of the burst.
+    slant_range_res : float
+        Slant range resolution of the burst.
+    shape : tuple
+        Tuples holding number of lines and samples of the burst.
+    az_time_interval : float
+        Azimth time interval of the burst.
+
+    Returns:
+    ------
+    _ : LUT2d
+        LUT2d calculated from poly1d.
+    '''
+    (n_lines, n_samples) = shape
+    # calculate all slant ranges in grid
+    slant_ranges = starting_slant_range + np.arange(n_samples) * slant_range_res
+
+    # no az dependency, but LUT2d required, so ensure all az coords covered
+    # TODO fix magic number of 2 days minus sensing_start
+    az_times = 2*24*3600 + np.array([0, n_lines * az_time_interval])
+
+    # calculate frequency for all slant range
+    freq_1d = np.array([doppler_poly1d.eval(t) for t in slant_ranges])
+
+    # init LUT2d (vstack freq since no az dependency) and return
+    return isce3.core.LUT2d(slant_ranges, az_times,
+                            np.vstack((freq_1d, freq_1d)))
+
+def get_burst_centers(tree):
+    '''
+    Parse grid points list and calculate burst center lat and lon
+
+    Parameters:
+    -----------
+    tree : Element
+        Element containing geolocation grid points.
+
+    Returns:
+    --------
+    _ : list
+        List of burst centers in degree longitude, latitude tuples.
+    _ : list
+        List of burst boundaries as list of degree longitude, latitude tuples.
+    '''
+    # find element tree
+    grid_pt_list = tree.find('geolocationGrid/geolocationGridPointList')
+
+    # read in all points
+    n_grid_pts = int(grid_pt_list.attrib['count'])
+    lines = np.empty(n_grid_pts)
+    pixels = np.empty(n_grid_pts)
+    lats = np.empty(n_grid_pts)
+    lons = np.empty(n_grid_pts)
+    for i, grid_pt in enumerate(grid_pt_list):
+        lines[i] = int(grid_pt[2].text)
+        pixels[i] = int(grid_pt[3].text)
+        lats[i] = float(grid_pt[4].text)
+        lons[i] = float(grid_pt[5].text)
+
+    unique_line_indices = np.unique(lines)
+    n_bursts = len(unique_line_indices) - 1
+    center_pts = [[]] * n_bursts
+    boundary_pts = [[]] * n_bursts
+
+    # zip lines numbers of bursts together and iterate
+    for i, (ln0, ln1) in enumerate(zip(unique_line_indices[:-1],
+                                       unique_line_indices[1:])):
+        # create masks for lines in current burst
+        mask0 = lines==ln0
+        mask1 = lines==ln1
+
+        # reverse order of 2nd set of points so plots of boundaries
+        # are not connected by a diagonal line
+        burst_lons = np.concatenate((lons[mask0], lons[mask1][::-1]))
+        burst_lats = np.concatenate((lats[mask0], lats[mask1][::-1]))
+
+        center_lon = np.mean(burst_lons)
```

Comment:
Displaying my ignorance here :D. Are there cases in which the burst geometry might not be regular (e.g. rectangle) and in these cases would it make sense to replace the mean with the centroid to compute "the centre" of the burst?

Reply:
I assume you're referring to mean vs centroid like [this](https://github.com/Turfjs/turf/issues/334)?

Reply:
Yes! I guess the two concepts should degenerate to the same thing for regular geometries (e.g. rectangles). This is not necessarily the case for irregular geometries. However, I think burst geometries should be consistently regular, so I guess having mean or centroid should not matter. Am I wrong?

Reply:
I don't think we'll encounter any irregular geometries so I agree with you.

What this doesn't address is rectangle intersects the date line. Do you know of a way to handle this?

Reply:
Excellent point. If we generate the burst polygon in lat/lon, we will run into the crossing date line issue. I think the correct way to handle it would be:
1. Check if the burst polygon crosses the dateline
2. If it does, split the polygon in two parts, one polygon before covering the burst area before the dateline, and another polygon covering the area after the dateline. 

What do you think?

Reply:
I will use [`check_dateline`](https://github.com/isce-framework/isce3/blob/e2c76c70c304a52fe583a0dd4e0455eab12475d9/python/packages/nisar/workflows/stage_dem.py#L39) from stage_dem.py as a guide.